tests/core/abi-utils/test_abi_utils.py

```diff
@@ -133,6 +165,634 @@ def test__abi_to_signature(abi, expected):
+        "stateMutability": "nonpayable",
+        "type": "function",
+    }
+
+
+@pytest.fixture
+def contract_abi() -> ABI:
+    return [
+        make_event_abi("LogNoArg"),
+        make_event_abi("LogSingleArg", [{"name": "arg0", "type": "uint256"}]),
+        make_event_abi("LogSingleWithIndex", [{"name": "arg0", "type": "uint256"}]),
+        make_function_abi("logTwoEvents", [{"name": "_arg0", "type": "uint256"}]),
+        make_function_abi("setValue", [{"name": "_arg0", "type": "uint256"}]),
+        make_function_abi(
+            "setValue",
+            [{"name": "_arg0", "type": "uint256"}],
+            [
+                {
+                    "name": "arg1",
+                    "components": [
+                        {"name": "a", "type": "uint256"},
+                        {"name": "b", "type": "uint256"},
+                    ],
+                }
+            ],
+        ),
+    ]
+
+
+@pytest.fixture
+def contract_abi_add_function() -> ABI:
+    return [
+        make_function_abi(
+            "add",
+            [{"name": "x", "type": "uint256"}, {"name": "y", "type": "uint256"}],
+            [{}],
+            [{"name": "sum", "type": "uint256"}],
+        ),
+    ]
+
+
+@pytest.fixture
+def contract_ambiguous_event() -> ABI:
+    return [
+        make_event_abi("LogSingleArg", [{"name": "arg0", "type": "uint256"}]),
+        make_event_abi("LogSingleArg", [{"name": "arg0", "type": "uint256"}]),
+    ]
+
+
+def test_get_all_event_abis(contract_abi) -> Sequence[ABIEvent]:
+    expected_event_abis = [
+        make_event_abi("LogNoArg"),
+        make_event_abi("LogSingleArg", [{"name": "arg0", "type": "uint256"}]),
+        make_event_abi("LogSingleWithIndex", [{"name": "arg0", "type": "uint256"}]),
+    ]
+    assert get_all_event_abis(contract_abi) == expected_event_abis
+
+
+def test_get_all_event_abis_without_events(
+    contract_abi_add_function,
+) -> Sequence[ABIEvent]:
+    assert get_all_event_abis(contract_abi_add_function) == []
+
+
+@pytest.mark.parametrize(
+    "event_name,input_args",
+    [
+        ("LogSingleArg", [{"name": "arg0", "type": "uint256"}]),
+        ("LogSingleWithIndex", [{"name": "arg0", "type": "uint256"}]),
+        ("LogNoArg", []),
+    ],
+)
+def test_get_event_abi(contract_abi, event_name, input_args):
+    expected_event_abi = make_event_abi(event_name, input_args)
+    input_names = [arg["name"] for arg in input_args]
+    assert get_event_abi(contract_abi, event_name, input_names) == expected_event_abi
+
+
+@pytest.mark.parametrize(
+    "name,args,error_type,expected_value",
+    (
+        (
+            None,
+            None,
+            ValidationError,
+            "event_name is required in order to match an event ABI.",
+        ),
+        ("foo", None, ValueError, "No matching events found"),
+    ),
+)
+def test_get_event_abi_raises_on_error(
+    contract_abi_add_function, name, args, error_type, expected_value
+):
+    with pytest.raises(error_type) as error:
```

Comment:
nit: you should be able to do something like: 
`with pytest.raises(error_type, match=expected_value)`
instead of the assertion on line 385